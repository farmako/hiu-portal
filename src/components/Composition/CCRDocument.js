import React from 'react';
import * as PropTypes from 'prop-types';
import CompositionComponent from './CompositionComponent';

const CCRDocument = ({ compositionData, consentReqId }) => {
  const compositions = compositionData.filter(node => node.resourceType.toLowerCase() === "composition");
  return compositions ? (
    compositions.map((compositon) => (
      <div style={{ marginBottom: 50 }}>
        <CompositionComponent
          composition={compositon}
          consentReqId={consentReqId}
          resources={compositionData}
        />
      </div>
    ))
  ) : (
    <div />
  );
};

const compositionDataShape = PropTypes.shape({
  resourceType: PropTypes.string,
  parentResources: PropTypes.arrayOf(PropTypes.shape({
    resourceType: PropTypes.string,
  })),
});

CCRDocument.propTypes = {
  compositionData: PropTypes.arrayOf(compositionDataShape).isRequired,
  consentReqId: PropTypes.string.isRequired
};

export default CCRDocument;
