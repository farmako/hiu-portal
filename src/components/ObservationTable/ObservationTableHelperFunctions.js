import React from 'react';
import valueForObs from './ObsValueHandlers';

export function getConceptText(codeableConcept) {
  if (codeableConcept && codeableConcept.text) {
    return codeableConcept.text;
  }
  if (codeableConcept && codeableConcept.coding) {
    return codeableConcept.coding[0].display;
  }

  return undefined;
}

export function getConceptTextForComponents(components) {
  const componentControls = [];
  if (components) {
    for (let i = 0; i < components.length; i++) {
      componentControls.push({
        text: getConceptText(components[i].code),
        value: valueForObs(components[i]),
      });
    }
    return (
      <ul>
        {componentControls.map((c) => (
          <li key={c.text}>
            {c.text}
            {' '}
            :
            {c.value}
          </li>
        ))}
      </ul>
    );
  }
  return undefined;
}

export const generateObservableEntity = (observation) => getConceptText(observation.code);

export const generateObservableEntityValue = (observation) => (
  <div>
    <div>{valueForObs(observation)}</div>
    {getConceptTextForComponents(observation.component)}
  </div>
);

export const generateObservableEntityStatusInterpretation = (observation) => (
  <div>
    {observation.status ? (
      <div>
        status:
        {observation.status}
      </div>
    ) : ''}
    {observation.interpretation ? (
      <div>
        interpretation:
        {' '}
        {getConceptText(observation.interpretation[0])}
      </div>
    ) : (
      ''
    )}
  </div>
);

export const generateRowsForMembers = (observations) => {
  const members = [];
  if (observations && observations[0].hasMember) {
    for (let i = 0; i < observations[0].resource.hasMember.length; i++) {
      const reference = observations[0].resource.hasMember[i].reference.split(
        '/',
      )[1];
      const referenceData = observations.find(
        (item) => item.resource.id === reference,
      ).resource;
      members.push(referenceData);
    }
    return members;
  }
  return undefined;
};
