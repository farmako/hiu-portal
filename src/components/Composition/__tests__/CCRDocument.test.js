import React from 'react';
import { shallow } from 'enzyme';
import CCRDocument from '../CCRDocument';
import ObservationTable from '../../ObservationTable/ObservationTable';
import MedicationRequestsComponent from '../../Medication/MedicationRequestsComponent';

describe('CCRDocument', () => {
  const mockData = [
    {
      resourceType: 'Composition',
      title: 'Discharge Summary',
      date: '2019-12-03',
      type: {
        coding: [
          {
            system: 'http://loinc.org',
            code: '28655-9'
          }
        ],
      },
      author: [
        {
          display: 'Dr.Divya'
        }
      ],
      event: [
        {
          period: {
            start: '2019-12-03',
            end: '2019-12-30'
          }
        }
      ],
      status: 'final',
    },
    {
      resourceType: 'MedicationRequest',
      parentType: 'Composition',
      status: 'active',
      intent: 'order',
      medicationReference: { reference: 'Medication/c2b8f3fd-6da0-44ed-bc91-8bb05ff614d2', display: 'IV antibiotics' },
      subject: { display: 'Shriya' },
      authoredOn: '2020-01-22',
      dosageInstruction: [],
    },
    {
      resourceType: 'Observation',
      parentType: 'Composition',
      status: 'active',
      intent: 'order',
      medicationReference: { reference: 'Medication/c2b8f3fd-6da0-44ed-bc91-8bb05ff614d2', display: 'IV antibiotics' },
      subject: { display: 'Shriya' },
      authoredOn: '2020-01-22',
      dosageInstruction: [],
    },
  ];
  const wrapper = shallow(
    <CCRDocument
      consentReqId="123"
      compositionData={mockData}
    />,
  );

  it('should render properly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
