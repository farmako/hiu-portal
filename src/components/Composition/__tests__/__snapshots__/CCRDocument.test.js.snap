// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CCRDocument should render properly 1`] = `
<div
  style={
    Object {
      "marginBottom": 50,
    }
  }
>
  <CompositionComponent
    composition={
      Object {
        "author": Array [
          Object {
            "display": "Dr.Divya",
          },
        ],
        "date": "2019-12-03",
        "event": Array [
          Object {
            "period": Object {
              "end": "2019-12-30",
              "start": "2019-12-03",
            },
          },
        ],
        "resourceType": "Composition",
        "status": "final",
        "title": "Discharge Summary",
        "type": Object {
          "coding": Array [
            Object {
              "code": "28655-9",
              "system": "http://loinc.org",
            },
          ],
        },
      }
    }
    consentReqId="123"
    resources={
      Array [
        Object {
          "author": Array [
            Object {
              "display": "Dr.Divya",
            },
          ],
          "date": "2019-12-03",
          "event": Array [
            Object {
              "period": Object {
                "end": "2019-12-30",
                "start": "2019-12-03",
              },
            },
          ],
          "resourceType": "Composition",
          "status": "final",
          "title": "Discharge Summary",
          "type": Object {
            "coding": Array [
              Object {
                "code": "28655-9",
                "system": "http://loinc.org",
              },
            ],
          },
        },
        Object {
          "authoredOn": "2020-01-22",
          "dosageInstruction": Array [],
          "intent": "order",
          "medicationReference": Object {
            "display": "IV antibiotics",
            "reference": "Medication/c2b8f3fd-6da0-44ed-bc91-8bb05ff614d2",
          },
          "parentType": "Composition",
          "resourceType": "MedicationRequest",
          "status": "active",
          "subject": Object {
            "display": "Shriya",
          },
        },
        Object {
          "authoredOn": "2020-01-22",
          "dosageInstruction": Array [],
          "intent": "order",
          "medicationReference": Object {
            "display": "IV antibiotics",
            "reference": "Medication/c2b8f3fd-6da0-44ed-bc91-8bb05ff614d2",
          },
          "parentType": "Composition",
          "resourceType": "Observation",
          "status": "active",
          "subject": Object {
            "display": "Shriya",
          },
        },
      ]
    }
  />
</div>
`;
